.github/actions/test-dotnet/Set-XUnitMaxParallelThreads.ps1

```diff
@@ -1,11 +1,24 @@
 param ($MaxParallelThreads)
 
-Write-Output "Replacing maxParallelThreads in xunit.runner.json files to $MaxParallelThreads."
+Write-Output "Setting maxParallelThreads in xunit.runner.json files to $MaxParallelThreads."
 
-($configFiles = Get-ChildItem "*/**/xunit.runner.json" -Recurse) | ForEach-Object {
-     (Get-Content $_) |
-        ForEach-Object  {$_ -Replace '"maxParallelThreads":\s*([-\d]*)', "`"maxParallelThreads`": $MaxParallelThreads"} |
-        Set-Content $_
+$parameters = @{
+        # -Filter is the fastest way to find files because its value is passed directly to the file system API.
+        Filter = "xunit.runner.json"
+        Recurse = $true
+        # -FollowSymLink is needed because without it errors will be thrown of type: 
+        # "Get-ChildItem: The system cannot find the path specified."
+        FollowSymlink = $true
+        # Ignore warnings that will be printed due to multiple symlinks pointing to the same physical file.
+        WarningAction = "Ignore"
+}
+
+($configFiles = Get-ChildItem @parameters) | ForEach-Object {
+    # Need the parentheses to close the file after reading. Without them, you'll receive an error with Set-Content:
+    # "The process cannot access the file '[...]\xunit.runner.json' because it is being used by another process."
```

Comment:
That's because the parentheses forces its contents to be evaluated completely so the cmdlet closes and the file handle is released before anything is passed onto the pipe. I think in this case it's both leaner and more appropriate to use `[IO.File]::ReadAllText($_)` and then there is no need for parentheses.

Reply:
I went with the native JSON approach.